File: app/src/main/java/it/scoppelletti/spaceship/cognito/sample/MainApp.java
package it.scoppelletti.spaceship.cognito.sample;

import android.app.Application;
import android.content.Context;
import com.amazonaws.regions.Regions;
import org.greenrobot.eventbus.EventBus;
import it.scoppelletti.spaceship.cognito.CognitoAdapter;
import it.scoppelletti.spaceship.security.SecureString;

public final class MainApp extends Application {
    
    @Override
    public void onCreate() {
        Context ctx;

        super.onCreate();

        EventBus.builder()
                .sendSubscriberExceptionEvent(false)
                .sendNoSubscriberEvent(false)
                .eventInheritance(false)
                .throwSubscriberException(BuildConfig.DEBUG)
                .addIndex(new MainEventBusIndex())
                .installDefaultEventBus();

        ctx = getApplicationContext();
        new CognitoAdapter.Builder(ctx)
                .poolId(BuildConfig.POOL_ID)
                .clientId(BuildConfig.CLIENT_ID)
                .clientSecret(new SecureString(BuildConfig.CLIENT_SECRET))
                .region(Regions.EU_WEST_1).build();
    }
}

File: spaceship-http-ui/app/src/main/kotlin/it/scoppelletti/spaceship/http/sample/MainApp.kt
@file:Suppress("unused")

package it.scoppelletti.spaceship.http.sample

import android.app.Application
import it.scoppelletti.spaceship.http.sample.inject.AppComponent
import it.scoppelletti.spaceship.http.sample.inject.DaggerAppComponent
import it.scoppelletti.spaceship.inject.UIComponent
import it.scoppelletti.spaceship.inject.UIComponentProvider
import it.scoppelletti.spaceship.inject.StdlibComponent
import it.scoppelletti.spaceship.inject.StdlibComponentProvider

class MainApp : Application(), UIComponentProvider,
        StdlibComponentProvider {

    private lateinit var _appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        _appComponent = DaggerAppComponent.factory()
                .create(this)
    }

    override fun uiComponent(): UIComponent = _appComponent

    override fun stdlibComponent(): StdlibComponent = _appComponent
}
